internal/internal_task_pollers.go

```diff
@@ -1369,3 +1376,21 @@ func convertActivityResultToRespondRequestByID(identity, domain, workflowID, run
 		Details:    details,
 		Identity:   common.StringPtr(identity)}
 }
+
+func emitHardwareMetricsTaggedScope(scope *metrics.TaggedScope) {
+	if scope != nil {
+		cpuPercent, _ := cpu.Percent(0, false)
+		cpuCores, _ := cpu.Counts(false)
+
+		scope.Gauge(metrics.NumCPUCores).Update(float64(cpuCores))
```

Comment:
Won't a guage get overridden unless it has a host-level dimension? If it does have a host-level dimension won't the cardinality be too much?

Reply:
@allenchen2244

Reply:
It is on datacenter-level dimension, I am currently looking at method to make it host-level